tests/routing.py

```diff
@@ -29,36 +29,73 @@ def twd_var(self,t=None):
     def getWindAt(self, t, lat, lon):
         return ( math.radians(self.twd_var(t)), self.tws_var(t), )
 
-def mock_point_validity(y,x):
-    return True
+class mock_point_validity:
+
+    def __init__(self, track, factor=4):
+        self.mean_point = ((track[0][1]+track[1][1])/2, (track[0][0]+track[1][0])/2)
+        self.mean_island = weatherrouting.utils.pointDistance(*(track[0]+track[1]))/factor
+
+    def point_validity(self, y, x):
+        if weatherrouting.utils.pointDistance(x,y,*(self.mean_point)) < self.mean_island:
+            return False
+        else:
+            return True
 
 polar_obj = weatherrouting.Polar(os.path.join(os.path.dirname(__file__),'Bavaria38.pol'))
 
-track = [(5,38),(5.2,38.2)]
 
 class TestRouting(unittest.TestCase):
 
     def setUp(self):
-        grib = mockGrib(2,180,0.1)
+        grib = mock_grib(2,180,0.1)
+        track = [(5,38),(5.2,38.2)]
+        island_route = mock_point_validity(track)
         self.routing_obj = weatherrouting.Routing(
             LinearBestIsoRouter,
             polar_obj,
             track,
             grib,
-            mock_point_validity,
+            island_route.point_validity,
             datetime.datetime.fromisoformat('2021-04-02T12:00:00')
         )
         
     def test_step(self):
         res = None 
         i = 0
-
+        
         while not self.routing_obj.end:
             res = self.routing_obj.step()
-            # print ("isocrones",res.isochrones)
-            # print("path",res.path)
-            # print("position",res.position)
             i += 1
-        
+
         self.assertEqual(i, 4)
         self.assertEqual(not res.path, False)
+
+
+class checkRoute(unittest.TestCase):
+
+    def setUp(self):
+        grib = mock_grib(4,225,0) 
+        self.track = track = [(5,38),(5.4,38.4)]
+        island_route = mock_point_validity(self.track, factor=4)
+        self.routing_obj = weatherrouting.Routing(
+            LinearBestIsoRouter,
+            polar_obj,
+            self.track,
+            grib,
+            island_route.point_validity,
+            datetime.datetime.fromisoformat('2021-04-02T12:00:00')
+        )
+        
+    def test_step(self):
+        res = None 
+        i = 0
+
+        while not self.routing_obj.end:
+            res = self.routing_obj.step()
+            print("step",i, res.progress)
+            i += 1
+        
+        path_to_end = res.path + [[*self.track[-1],'']]
+        print (weatherrouting.utils.pathAsGeojson(path_to_end))
```

Comment:
Please comment out all print function, since they're only for debug, or use logging instead like this, so it can be turned on / off with one line of code:

```
log = logging.getLogger('test_2')

log.debug ('What you want to print for debugging')
```